debezium-core/src/main/java/io/debezium/relational/TableId.java

```diff
@@ -36,11 +32,14 @@ public static TableId parse(String str) {
      * @param useCatalogBeforeSchema {@code true} if the parsed string contains only 2 items and the first should be used as
      *            the catalog and the second as the table name, or {@code false} if the first should be used as the schema and the
      *            second as the table name
+     * @param identifierTransformer transformer used for unquoting the identifier parts in the given string
      * @return the table ID, or null if it could not be parsed
      */
-    public static TableId parse(String str, boolean useCatalogBeforeSchema) {
-        String[] parts = IDENTIFIER_SEPARATOR.split(str);
-        if ( parts.length < 0 ) return null;
+    public static TableId parse(String str, boolean useCatalogBeforeSchema, IdentifierTransformer identifierTransformer) {
```

Comment:
Why do we even have to identify the quoting character? Seems like by default we should just be able to understand the single and double quoted identifiers.